app/src/openxr/cpp/OpenXRInput.cpp

```diff
@@ -58,16 +60,15 @@ XrResult OpenXRInput::Initialize(ControllerDelegate& delegate)
     }
   }
 
-  bool supportsEyeGazeExtension = OpenXRExtensions::IsExtensionSupported(XR_EXT_EYE_GAZE_INTERACTION_EXTENSION_NAME);
-  if (supportsEyeGazeExtension)
+  if (isEyeTrackingSupported)
     InitializeEyeGaze(delegate);
 
   XrSessionActionSetsAttachInfo attachInfo { XR_TYPE_SESSION_ACTION_SETS_ATTACH_INFO };
```

Comment:
If I've understood it correctly, it's mandatory that we attach the action sets to the session before executing the spaces initialization; if that's the case, I don't see this requirement as obvious, looking at the InitializeEyeGazeSpace function's code. In that case, I believe a code comment would be useful.

Reply:
Not really, the thing is that we need to create the actions before attaching them which is what xrAttachSessionActionSets does bellow.